rfc/networked-dot-storage.md

```diff
@@ -0,0 +1,174 @@
+
+One proposed path forward is to not pursue "decentralization" beyond trying to make other nodes a reality, perhaps through incentivization, and leaning heavily into product improvements that differentiate our platform. We could aim for "S3 if AWS invested in open source S3 implementations and made it content addressed".
+
+This is not the path we are going to pursue. It feels important to say that up front.
+
+But before we go deeper into where we are headed, it's worth looking at the implied dichotomy between "decentralization" and product features. That this dichotomy is almost reflexively referred to likely results from failing to appropriately frame "decentralization" in terms of product features. So let's start by framing the product features we want to unlock through decentralization.
+
+## The Network We Want
+
+### No Data Loss
+
+1. Users should be able to specify how many unique copies of their data they want backed up, and the W3S network should handle insuring that many copies remain backed up as long as the user pay to store in W3S.
+
+### Perpetuity
+
+1. Users should be able to store hot data in W3S that will be available for  retrieval until they stop paying for it. They should not have to worry about any single business entity holding their data disappearing. If it does, the data should go somewhere else without their needing to intervene.
+
+### Sensible defaults with customization
+
+1. Users should have the option to store with simply "the W3S network" rather than a specific business entity or Filecoin SP. This is a hard lesson learned from Filecoin that forcing users to choose from a marketplace where information on available options is hard to come by is a terrible user experience.
+
+2. Users should still have the option to choose a provider they like, or to move between providers.
+
+### Trust
+
+In general, users should be able to "trust" W3S to handle their data well and meet service level agreements. That trust should be derived by giving users the ability to verify any actions taken on the W3S network. 
+
+1. Users should be able to gather a detailed history of what has happened with their data when they put it on the W3S Network.
+
+2. A user paying for storage should be able to verify that data is being stored.
+
+3. A user paying for storage should be able to verify it can be retrieved quickly. 
+
+4. A user paying for multiple backups should be able to verify each backup is still functioning.
+
+4. If verification demonstrates that a particular provider is failing to provide a service that the user has paid for, the network should detect the failures and move their data automatically. (or alternatively, the user should have the ability to create a dispute and move to a different provider while getting money back)
```

Comment:
```suggestion
5. If verification demonstrates that a particular provider is failing to provide a service that the user has paid for, the network should detect the failures and move their data automatically. (or alternatively, the user should have the ability to create a dispute and move to a different provider while getting money back)
```

Reply:
I liked the idea @Gozala pointed out of both the service, as well as the providers should put some collateral that can be collected when they fail to meet their promises

Reply:
> I liked the idea @Gozala pointed out of both the service, as well as the providers should put some collateral that can be collected when they fail to meet their promises

@vasco-santos I am not sure if you read the crypto economic requirements proposal. This contains the collateral requirement. We will need this for all the nodes (storage, retrieval, index)

Reply:
oh cool, no I did not know about that requirements document IIRC

Reply:
this sounds good! 

fwiw slashing mechanisms always make me nervous because they open up a big new vector for attacks on the network, ie, malicious actors creating conditions where a network participant can be slashed "unfairly" - for example by DDOSing a single network participant with the explicit goal of hurting them and the larger network

I've read compelling arguments that slashing can actually make a network more fragile, but unfortunately can't find it right now - in lieu of that here are two takes from Emin Gun Sirer (founder of Avalanche and arguably the inventor of proof-of-work) and Anatoly Yakovenko that point in that direction (apologies for linking to the hellsite):

https://twitter.com/el33th4xor/status/1294773537128034304
https://twitter.com/aeyakovenko/status/1295357700570660865

This doesn't mean that slashing is bad for our system, but imho it introduces a LOT of additional complexity for questionable value.

Reply:
this at least gets into some of the complexity here: https://a16zcrypto.com/posts/article/the-cryptoeconomics-of-slashing/